### PR TITLE
change to repo.osgeo.org

### PR DESCRIPTION
**Pull request by jodygarnett**
This project makes use of geoserver repository for release publication (mirroring what was done for repo.boundlessgeo.com).

Signed-off-by: Jody Garnett <jody.garnett@gmail.com>

### CODE REVIEW

**Comment by jodygarnett** [comment]
@tbarsballe following on from https://github.com/geotools/geotools/pull/2863 here is a change to geowebcache.

Please note that each project has a different distribution management repository to upload to (in this case geowebcache is sharing with geoserver to better mirror what was done at repo.boundlessgeo.com).

File: geowebcache/pom.xml [hunk 1]
```diff
@@ -53,32 +53,43 @@
   
   <repositories>
     <repository>
-      <id>boundless</id>
-      <name>Boundless Maven Repository</name>
-      <url>https://repo.boundlessgeo.com/main/</url>
+      <id>osgeo-release</id>
+      <name>OSGeo Repository</name>
```
Comment (tbarsballe) [inline geowebcache/pom.xml:57]: Maybe use "OSGeo Release Repository: for the name here (and "OSGeo Snapshot Repository" below)

File: geowebcache/pom.xml [hunk 2]
```diff
@@ -53,32 +53,43 @@
...
-      <id>osgeo</id>
-      <name>Open Source Geospatial Foundation Repository</name>
-      <url>https://download.osgeo.org/webdav/geotools/</url>
+      <id>osgeo-snapshot</id>
+      <name>OSGeo Repository</name>
+      <url>https://repo.osgeo.org/repository/snapshot/</url>
+      <!-- Contains releases for OSGeo projects: GeoTools, GeoWebCache, GeoServer, ...     -->
+      <!-- Includs third-party-dependences, and caches of:                                 -->
+      <!-- ucar (https://artifacts.unidata.ucar.edu/content/repositories/unidata-releases) -->
+      <!-- geosolutions (http://maven.geo-solutions.it/)                                   -->
```
Comment (tbarsballe) [inline geowebcache/pom.xml:73]: Aren't these only in the release repository, not the snapshot one?

Also, I don't believe GeoWebCache cares about either of those (maybe geosolutions?), so there isn't much of a reason to mention them

**Comment by tbarsballe** [comment]
GeoTools builds have completed successfully on Jenkins, and geotools-snapshots is now populated with 24-SNAPSHOT on nexus.

All build checks have passed here, and the only issues I had are with docs/naming, so feel free to merge this when ready.

**Comment by tbarsballe** [comment]
New failure, so the GeoTools snapshots are working:
```
[ERROR] Failed to execute goal on project gs-importer-rest: Could not resolve dependencies for project org.geoserver.importer:gs-importer-rest:jar:2.18-SNAPSHOT: Could not find artifact org.restlet:org.restlet.ext.fileupload:jar:1.0.8 in osgeo-releases (https://repo.osgeo.org/repository/release/)
```

Looks like we should be mirroring Restlet as well, I'll look into it...

**Pull request merged**